### ISSUE TITLE
Examples should not add program options multiple times

### ISSUE THREAD

**Issue by HadrienG2** [acts-project/acts#483]
Some Acts examples involuntarily add certain program options multiple times. For example, the `ActsExampleFatrasAligned` example adds the magnetic field options twice, [once directly as part of the general Fatras options setup in `fatrasMain()`](https://github.com/acts-project/acts/blob/88981f64cb1a7c1f234806e0755dbc7660603f65/Examples/Run/Fatras/Common/FatrasMain.cpp#L46) and once as part of [the `AlignedDetector` options setup](https://github.com/acts-project/acts/blob/88981f64cb1a7c1f234806e0755dbc7660603f65/Examples/Detectors/ContextualDetector/src/AlignedDetector.cpp#L30) which [is also triggered by `fatrasMain()`](https://github.com/acts-project/acts/blob/88981f64cb1a7c1f234806e0755dbc7660603f65/Examples/Run/Fatras/Common/FatrasMain.cpp#L54).

It would be really nice if this worked, because it would allow us to seamlessly specify interdependencies between Acts program options (e.g. "if you want `AlignedDetector` options, you also want BField options"). But this is a Boost library that we are talking about, so obviously anything beyond the basics is broken, and if you do this you will get a weird error:

```
$ ../spack-build/bin/ActsExampleFatrasAligned -n100 --output-dir=sim-gun --output-csv=1 --evg-input-type=gun --bf-values=0 0 2
terminate called after throwing an instance of 'boost::wrapexcept<boost::program_options::ambiguous_option>'
  what():  option '--bf-values=0' is ambiguous and matches different versions of '--bf-values'
Aborted (core dumped)
```

I think the least bad option here might be to do the dependency tracking ourselves by tracking which program options have already been added, either in some kind of `boost::program_options::options_description` wrapper or in a global variable if you're feeling dirty. But I'm curious to see what other opinions on this topic are. After all, another point of view would be to say that `AlignedDetector` should not add BField options and leave the whole dependency management work to the top-level example, in this case Fatras.

**Comment by HadrienG2** [comment]
Still a problem.

**Comment by asalzburger** [comment]
I see, I will assign to me.

**Comment by HadrienG2** [comment]
This seems resolved as of master 5038ab6070b8eb89966a80127816bb961c40c8c4 ! :tada: 

At least the original reproducer does not work anymore, after adjusting it to the new CLI flag syntax.
